docs/api_reference/src/quantities.tex

```diff
@@ -30,72 +35,581 @@
 \rSec1[mp.units.core.syn]{Module \tcode{mp_units.core} synopsis}
 \indexmodule{mp_units.core}%
 \begin{codeblock}
+// mostly freestanding
 export module mp_units.core;
 
 import std;
 
 export namespace mp_units {
 
-export enum class quantity_character { scalar, vector, tensor };
+// \ref{qty.helpers}, helpers
+
+// \ref{qty.sym.txt}, class template \tcode{symbol_text}
+
+enum class @\libglobal{character_set}@ : std::int8_t { utf8, portable, default_character_set = utf8 };
+
+template<std::size_t N, std::size_t M>
+class symbol_text;
+
+// \ref{qty.sym.expr}, symbolic expressions
+
+// \ref{qty.sym.expr.types}, types
+
+template<typename T, typename... Ts>
+struct per;
+
+template<typename F, int Num, int... Den>
+  requires @\seebelownc@
+struct power;
+
+// \ref{qty.ref}, reference
+
+// \ref{qty.dim}, dimension
+
+// \ref{qty.dim.concepts}, concepts
+
+template<typename T>
+concept Dimension = @\seebelownc@;
+
+template<typename T, auto D>
+concept DimensionOf = @\seebelownc@;
+
+// \ref{qty.dim.types}, types
+
+template<symbol_text Symbol>
+struct base_dimension;
+
+template<@\exposconceptnc{SymbolicConstant}@... Expr>
+struct derived_dimension;
+
+struct dimension_one;
+inline constexpr dimension_one @\libglobal{dimension_one}@{};
+
+consteval @\libconcept{Dimension}@ auto @\liboverload{inverse}{\cname{Dimension}}@(@\libconcept{Dimension}@ auto d) { return dimension_one / d; }
+
+template<std::intmax_t Num, std::intmax_t Den = 1, @\libconcept{Dimension}@ D>
+  requires(Den != 0)
+consteval @\libconcept{Dimension}@ auto pow(D d);
+consteval @\libconcept{Dimension}@ auto @\liboverload{sqrt}{\cname{Dimension}}@(@\libconcept{Dimension}@ auto d) { return pow<1, 2>(d); }
+consteval @\libconcept{Dimension}@ auto @\liboverload{cbrt}{\cname{Dimension}}@(@\libconcept{Dimension}@ auto d) { return pow<1, 3>(d); }
```

Comment:
Why are these not part of _`dimension-interface`_?
The interface of a dimension is divided
between these functions and those in _`dimension-interface`_.

It is typical for an all-encompassing domain concept,
such as `Dimension`, `QuantitySpec`, and `Unit`,
to have its interface divided between `friend`s in a class scope
and non-`friend` non-member functions.

I'd like to be able to describe such interfaces all next to each other.

Reply:
I had some implementability issues (maybe because of `dimension_one`). If you can refactor code to work OK, then I am strongly in favor of such a change.

Reply:
What I meant is that I had problems compiling such code.

Reply:
Yeah, I know.
But I won't be touching actualy code for the time being.